docs/_data/nakadi-event-bus-api.yaml

```diff
@@ -2469,6 +2478,56 @@ definitions:
 
           If this is set it MUST be a valid required field as defined in the schema.
 
+      cleanup_policy:
+        type: string
+        x-extensible-enum:
+          - delete
+          - compact
+        default: 'delete'
+        description: |
+          Event type cleanup policy. There are two possible values:
+
+          - 'delete' cleanup policy will delete old events after retention time expires. Nakadi guarantees that each
+            event will be available for at least the retention time period. However Nakadi doesn't guarantee that event
+            will be deleted right after retention time expires.
+
+          - 'compact' cleanup policy will keep only the latest event for each event key. The compaction is performed per
+            partition, there is no compaction across partitions. The key that will be used as a compaction key should be
+            specified in 'partition_compaction_keys' field.
+
+            The compaction can be not applied to events that were published recently and located at the head of the
+            queue, which means that the actual amount of events received by consumers can be different depending on time
+            when the consumption happened.
+
+            When using 'compact' cleanup policy user should consider that different Nakadi endpoints showing the amount
+            of events will actually show the original amount of events published, not the actual amount of events that
+            are currently there.
+            E.g. subscription /stats endpoint will show the value 'unconsumed_events' - but that may not match with the
+            actual amount of events unconsumed in that subscription as 'compact' cleanup policy may delete older events
+            in the middle of queue if there is a newer event for the same key published.
+
+            For more details about compaction implementation please read the documentation of Log Compaction in Kafka
+            https://kafka.apache.org/documentation/#compaction, Nakadi currently relies on this implementation.
```

Comment:
Should we refer to Kafka-specific details in Nakadi? One of the advantages of Nakadi is that it can (only potentially so far, but still) support multiple storages. This looks a bit like an abstraction leak

Reply:
Good point.  I think it should be rephrased, less direct. It is fine only as an example. "For example, you can read the documentation of Log Compaction in Kafka"

Reply:
@lmontrieux if we are not referering to Kafka I would argue that we should rewrite the whole section about log compaction, with diagrams and all. Otherwise it will not be possible to understand how the feature works as it works. We would also not be able to refer to terminology specific from Kafka, like "head of the queue".

Reply:
@lmontrieux @SergKam I don't agree that it's a good point. It's like talking about a chicken without saying the word chicken. Like, it has feathers, eat corn and put eggs. In this API we need to use the concept of "head queue" and "head tail" which is defined only in Kafka docs. We would have to redefine them here if we want to mention these terms. They are not self explanatory. Putting myself in the shoes of someone who do not not know how log compaction works, if I do not read Kafka docs, I would have a thousand questions. We either follow the pragmatic approach of explicitly pointing to Kafka docs or we exhaustively explain how it works in our docs.

Reply:
One option would be to literaly copy and paste Kafka docs into Nakadi and adjust the wording, if that doesn't break with any laws.

Reply:
@rcillo My proposal was to keep the link to Kafka docs and everything related, but just to mention, that it can be another backend system supporting compaction. like https://github.com/apache/incubator-pulsar/wiki/PIP-14:-Topic-compaction